server/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregatorTestCase.java

```diff
@@ -191,11 +213,45 @@ public void testGeoShapeWithSeveralDocs() throws IOException {
                 assertEquals((long) expectedCountPerGeoHash.get(bucket.getKeyAsString()), bucket.getDocCount());
             }
             assertTrue(AggregationInspectionHelper.hasValue(geoHashGrid));
-        }, new GeoShapeFieldMapper.GeoShapeFieldType());
+        }, new GeoShapeFieldMapper.GeoShapeFieldType(), new NoneCircuitBreakerService());
+    }
+
+    public void testGeoShapeTrippedCircuitBreaker() throws IOException {
+        GeoGridTiler tiler = geoGridTiler();
+        int precision = randomIntBetween(1, 9); // does not go until MAX_ZOOM for performance reasons
```

Comment:
Just verifying this won't nuke CI, since it has to run at least once to determine the number tiles?  I'm assuming that's why it's limited to precision: 9 but thought I'd double check :)

Reply:
yeah, it is just that I am not interested in tiling a shape into 100s of thousands of tiles. I've run this test over 1000 iterations and things seem to run snappily

Reply:
:+1:

Reply:
:+1: